File: backend/modules/Portfolio/CromaticaCreativa.Modules.Portfolio.Infrastructure/Persistence/Mappers/CorporateClientMapper.cs
using CromaticaCreativa.Modules.Portfolio.Domain.Aggregates;
using CromaticaCreativa.Modules.Portfolio.Domain.Enums;
using CromaticaCreativa.Modules.Portfolio.Domain.ValueObjects;
using CromaticaCreativa.Modules.Portfolio.Infrastructure.Persistence.Models;

namespace CromaticaCreativa.Modules.Portfolio.Infrastructure.Persistence.Mappers;

public static class CorporateClientMapper
{
    public static CorporateClient ToDomain(CorporateClientModel model)
    {
        ArgumentNullException.ThrowIfNull(model);

        var client = CorporateClient.Create(
            new CorporateClientId(model.Id),
            new CorporateClientName(model.Name),
            new MediaReference(model.LogoReference));

        switch (model.VisibilityStatus)
        {
            case PortfolioPersistenceValues.Visible:
                client.Show();
                break;
            case PortfolioPersistenceValues.Hidden:
                client.Hide();
                break;
            default:
                throw new InvalidOperationException(
                    $"Corporate client '{model.Id}' has unsupported visibility status '{model.VisibilityStatus}'.");
        }

        return client;
    }

    public static CorporateClientModel ToPersistence(CorporateClient client)
    {
        ArgumentNullException.ThrowIfNull(client);

        return new CorporateClientModel
        {
            Id = client.Id.Value,
            Name = client.Name.Value,
            LogoReference = client.Logo.Value,
            VisibilityStatus = client.Visibility switch
            {
                VisibilityStatus.Visible => PortfolioPersistenceValues.Visible,
                VisibilityStatus.Hidden => PortfolioPersistenceValues.Hidden,
                _ => throw new InvalidOperationException($"Unsupported Domain visibility status '{client.Visibility}'.")
            }
        };
    }
}

File: backend/src/modules/Portfolio/Portfolio.Infrastructure/Persistence/Mappers/CorporateClientMapper.ts
import { CorporateClientPersistenceDto } from '../../../Portfolio.Commons/DTOs/CorporateClientPersistenceDto';
import { CorporateClient } from '../../../Portfolio.Domain/Aggregates/CorporateClient';
import { VisibilityStatus } from '../../../Portfolio.Domain/Enums/VisibilityStatus';
import { CorporateClientId } from '../../../Portfolio.Domain/ValueObjects/CorporateClientId';
import { CorporateClientName } from '../../../Portfolio.Domain/ValueObjects/CorporateClientName';
import { MediaReference } from '../../../Portfolio.Domain/ValueObjects/MediaReference';
import { CorporateClientPersistenceModel } from '../Models/CorporateClientPersistenceModel';

export class CorporateClientMapper {
  public static toDomain(model: CorporateClientPersistenceDto): CorporateClient {
    const client = CorporateClient.create(new CorporateClientId(model.id), new CorporateClientName(model.name),
      new MediaReference(model.logoReference));
    if (model.visibilityStatus === VisibilityStatus.Visible) client.show();
    else if (model.visibilityStatus === VisibilityStatus.Hidden) client.hide();
    else throw new Error(`El cliente corporativo '${model.id}' tiene la visibilidad no soportada '${model.visibilityStatus}'.`);
    return client;
  }

  public static toPersistence(client: CorporateClient): CorporateClientPersistenceModel {
    return Object.assign(new CorporateClientPersistenceModel(), {
      id: client.id.value,
      name: client.name.value,
      logoReference: client.logo.value,
      visibilityStatus: client.visibility,
      projects: [],
    });
  }
}
